presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -531,7 +531,7 @@ folly::SemiFuture<std::unique_ptr<HttpResponse>> HttpClient::sendRequest(
     sendRequest(std::move(responseHandler));
   };
 
-  if (delayMs > 0) {
+  if (delayMs > 0 && eventBase_ != nullptr) {
```

Comment:
else condition uses eventBase_ as well. We should make sure life time of event base be equal or longer than client.

Reply:
Yeah you are spot on! Let me fix it.